src/context/TransactionsProvider/TransactionsProvider.tsx

```diff
@@ -83,6 +71,15 @@ export const TransactionsProvider = ({ children }: TransactionsProviderProps): J
               },
               (err: any) => console.error(err)
             )
+            if (isEmpty(accountSpecifiers)) continue
+            accountSpecifiers.forEach(accountSpecifierMap => {
+              dispatch(
+                txHistoryApi.endpoints.getAllTxHistory.initiate(
+                  { accountSpecifierMap },
```

Comment:
see the logic in transactions provider of how we narrow this full `accountSpecifiers` array down to the individual map that's required, we don't have to fetch all of these restfully yet

Reply:
Reworked the call to move outside the inner loop and now getting account specifiers by caip2 and fetching tx history based on the caip2's account specifiers.